pkg/k8s/operator/status/agent_status_manager.go

```diff
@@ -116,12 +117,20 @@ func (a *agentStatusManager) getDaemonSet(ctx context.Context) result.Result[ins
 }
 
 func (a *agentStatusManager) getConfigSecret(ctx context.Context) result.Result[instanav1.ResourceInfo] {
+	if !objectKeyHasName(a.agentSecretConfig) {
```

Comment:
It's easier to reason on non-negated conditions, so having `objectKeyDoesHasNoName` or `objectHasNoName` is simpler to read and reason and can be done by simply reversing the condition checked in the function for `key.Name == ""`